### ISSUE TITLE
no-unused-vars only generates one error even if multiple trailing arguments are unused

### ISSUE THREAD

**Issue by Niggler** [eslint/eslint#9875]
**Tell us about your environment** reproducible on the demo page at https://eslint.org/demo/

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:** default from demo page (`no-unused-vars` true)

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function f(x,y,z,w) { return y; } f(1,2);
```

**What did you expect to happen?**

Since `no-unused-vars` is enabled, two errors should be generated:

```
1:16 - 'z' is defined but never used
1:18 - 'w' is defined but never used
```

**What actually happened? Please include the actual, raw output from ESLint.**

Only the second error is generated.  Despite the fact that `z` is unused and every argument after the `z` is unused, eslint does not generate an error for `z`.  It only shows up after fixing the error with `w`.

**Comment by platinumazure** [comment]
Closing as a duplicate of #9750. (That said, thank you for creating this issue, as it reminded me of the need to drum up support from the team to move #9750 forward.)
